api/reports.go

```diff
@@ -0,0 +1,133 @@
+package api
+
+import (
+	"memberserver/api/models"
+	"memberserver/datastore"
+	"net/http"
+	"sort"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type ReportsServer struct {
+	store datastore.ReportStore
+}
+
+func (r *ReportsServer) GetMemberCounts(w http.ResponseWriter, req *http.Request) {
+	chartType := req.URL.Query().Get("type")
+	month := req.URL.Query().Get("month")
+
+	if len(month) > 0 {
+		date, err := time.Parse("", month)
+		if err != nil {
+			http.Error(w, "error looking up counts by month - use a valid date", http.StatusNotFound)
+			return
+		}
+		ok(w, makeDistritutionChartByMonth(date, r.store))
+		return
+	}
+
+	var charts []models.ReportChart
+	memberCounts, err := r.store.GetMemberCounts()
+	if err != nil {
+		http.Error(w, "error getting member counts", http.StatusNotFound)
+		return
+	}
+
+	if len(chartType) > 0 {
+		if chartType == "line" {
+			charts = append(charts, makeMemberTrendChart(memberCounts))
+		}
+
+		if chartType == "pie" {
+			charts = makeMemberDistributionChart(memberCounts)
+		}
+	}
+
+	if len(chartType) == 0 {
+		charts = append(charts, makeMemberTrendChart(memberCounts))
+		charts = append(charts, makeMemberDistributionChart(memberCounts)...)
+	}
+
+	ok(w, charts)
+}
+
+func makeMemberTrendChart(counts []models.MemberCount) models.ReportChart {
+	var chart models.ReportChart
+	chart.Options.Title = "Membership Trends"
+	chart.Type = "line"
+	chart.Options.CurveType = "function"
+	chart.Options.Legend = "bottom"
+	chart.Cols = []models.ChartCol{{Label: "Month", Type: "string"}, {Label: "Member Count", Type: "number"}}
+
+	sort.Slice(counts, func(i1, i2 int) bool {
```

Comment:
Why do the sort here? Wouldn't it be better to use sql `orderBy` to sort? A DB sort is more efficient compared to sorting it in the actual codebase